frontend/src/lib/components/SaveToCohortModalContent/saveToCohortModalContentLogic.ts

```diff
@@ -0,0 +1,123 @@
+            DEFAULT_COHORT_FILTERS,
+            {
+                setCohortFilters: (state, { filters }) => {
+                    return { ...state, ...filters }
+                },
+            },
+        ],
+    }),
+    loaders(({ values }) => ({
+        cohorts: {
+            __default: { count: 0, results: [] } as CountedPaginatedResponse<CohortType>,
+            loadCohorts: async () => {
+                const response = await api.cohorts.listPaginated({
+                    ...values.paramsFromFilters,
+                })
+                return {
+                    count: response.count,
+                    results: response.results,
+                }
+            },
+        },
+    })),
+    selectors(({ actions }) => ({
+        count: [(selectors) => [selectors.cohorts], (cohorts) => cohorts.count],
+        paramsFromFilters: [
+            (s) => [s.cohortFilters],
+            (filters: CohortFilters) => ({
+                ...filters,
+                limit: COHORTS_PER_PAGE,
+                offset: filters.page ? (filters.page - 1) * COHORTS_PER_PAGE : 0,
+            }),
+        ],
+        pagination: [
+            (s) => [s.cohortFilters, s.count],
+            (filters, count): PaginationManual => {
+                return {
+                    controlled: true,
+                    pageSize: COHORTS_PER_PAGE,
+                    currentPage: filters.page || 1,
+                    entryCount: count,
+                    onBackward: () => actions.setCohortFilters({ page: filters.page != null ? filters.page - 1 : 1 }),
+                    onForward: () => actions.setCohortFilters({ page: filters.page != null ? filters.page + 1 : 1 }),
+                }
+            },
+        ],
+    })),
+    listeners(({ actions, values }) => ({
+        setCohortFilters: async () => {
+            actions.loadCohorts()
+        },
+        saveQueryToCohort: async ({ cohort, query }) => {
+            const toastId = 'toast-' + Math.random()
+            try {
+                lemonToast.info('Saving cohort...', { toastId, autoClose: false })
+                await api.update(`api/projects/${values.currentProjectId}/cohorts/${cohort.id}`, {
+                    query: query,
+                })
+
+                const mountedCohortEditLogic = cohortEditLogic.findMounted({ id: cohort.id })
+                await mountedCohortEditLogic?.actions.updateCohortCount()
+
+                await delay(500) // just in case the toast is too fast
+                lemonToast.dismiss(toastId)
+                lemonToast.success('Cohort saved', {
+                    toastId: `${toastId}-success`,
+                    button: {
+                        label: 'View cohort',
+                        action: () => router.actions.push(urls.cohort(cohort.id)),
+                    },
+                })
+            } catch {
+                lemonToast.dismiss(toastId)
+                lemonToast.error('Cohort save failed')
```

Comment:
Might make sense to log the error to the console so we can see more details.

```suggestion
                lemonToast.error('Cohort save failed')
                console.error('Failed to save cohort:', error)
```